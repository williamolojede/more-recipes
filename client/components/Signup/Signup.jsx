import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import Footer from '../shared/Footer.jsx';
import AuthForm from '../shared/AuthForm.jsx';
import Preloader from '../shared/Preloader.jsx';
import ErrorDisplay from '../shared/ErrorDisplay.jsx';

class Signup extends Component {
  render() {
    const { isFetching, errorMessage, isAuthenticated, location } = this.props;
    const { from } = location.state || { from: { pathname: '/' } };
    if (isAuthenticated) {
      return (
        <Redirect to={from} />
      );
    }

    return (
      <div className="page page__register">
        <header className="header center">
          <a href="/" className="white-text brand-logo">MoreRecipes</a>
        </header>
        <main>
          <div className="container">
            <div className="row">
              <div className="register-card card col s11 m6">
                <div className="card-content">
                  {
                    errorMessage ? <ErrorDisplay message={errorMessage} /> : null
                  }
                  <h1 className="card-title center">Create a MoreRecipes Account</h1>
                  <AuthForm type="signup" />
                </div>
              </div>
            </div>
            <p className="white-text center">
              Donâ€™t have an account?
              <Link to="/login" className="white-text"> Register now</Link>
            </p>
            {
              isFetching ? <Preloader /> : null
            }
          </div>
        </main>
        <Footer />
      </div>
    );
  }
}

Signup.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string,
  location: PropTypes.object,
  isAuthenticated: PropTypes.bool.isRequired
};

const mapStateToProps = ({ auth }) => {
  const { isAuthenticated, errorMessage, user, isFetching } = auth;
  // whichever is undefined thats is not in auth wont be added to the props
  return {
    isAuthenticated,
    errorMessage,
    user,
    isFetching
  };
};

export default connect(mapStateToProps)(Signup);

