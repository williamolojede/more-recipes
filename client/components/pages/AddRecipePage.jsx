import React from 'react';

import ManageRecipes from '../shared/ManageRecipes';
import AddRecipeForm from '../shared/AddRecipeForm';

const AddRecipePage = props => (
  <ManageRecipes>
    <AddRecipeForm isEditMode={false} {...props} />
  </ManageRecipes>
);
export default AddRecipePage;
