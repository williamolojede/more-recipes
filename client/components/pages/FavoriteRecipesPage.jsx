import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import swal from 'sweetalert';


import decodeToken from '../../utils/decodeToken';

import ManageRecipes from '../shared/ManageRecipes';
import Preloader from '../shared/Preloader';
import RecipeCardShort from '../shared/RecipeCard/RecipeCardShort';

import { fetchFavoriteRecipes, favorite } from '../../actions/recipe';

class FavoriteRecipesPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipes: [],
      currentPage: 1,
      limit: 12,
      last: 1,
    };

    this.payload = {
      userId: decodeToken().id,
      page: this.state.currentPage,
      limit: this.state.limit
    };
  }

  componentWillMount() {
    const payload = {
      userId: decodeToken().id,
      page: this.state.currentPage,
      limit: this.state.limit
    };
    this.props.dispatch(fetchFavoriteRecipes(payload));
  }

  componentWillReceiveProps(newProps) {
    if (newProps.recipes) {
      const {
        recipes,
        pagination
      } = newProps;
      this.setState({
        recipes,
        last: pagination.last
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.currentPage !== this.state.currentPage) {
      const payload = {
        userId: decodeToken().id,
        page: this.state.currentPage,
        limit: this.state.limit
      };
      this.props.dispatch(fetchFavoriteRecipes(payload));
    }
  }

  handlePageClick = ({ selected }) => {
    this.setState({
      currentPage: selected + 1
    });
  }

  removeRecipe = (id) => {
    swal({
      title: 'Remove recipe from your favorite List?',
      icon: 'warning',
      buttons: true,
      dangerMode: true,
    })
      .then((willDelete) => {
        if (willDelete) {
          return this.props.dispatch(favorite(id));
        }
        return willDelete;
      })
      .then((willDelete) => {
        if (!willDelete) return;
        const payload = {
          userId: decodeToken().id,
          page: (this.state.recipes.length === 1 && this.state.last !== 1) ?
            this.state.currentPage - 1 : this.state.currentPage,
          limit: this.state.limit
        };
        this.props.dispatch(fetchFavoriteRecipes(payload));
      });
  }

  renderBody = () => {
    const { isFetching } = this.props;

    if (isFetching) {
      return (<Preloader />);
    } else if (!isFetching && this.state.recipes.length === 0) {
      return (
        <div className="no-favorites">
          <span>Welcome to your favorites!</span>
          <span>Tap <i className="mdi mdi-heart" /> to save any recipe you like,</span>
          <span>and it&apos;ll show up here.</span>
        </div>
      );
    }
    return (
      <ul className="row recipes-list">
        {
          this.state.recipes.map(({ id, name }) => (<RecipeCardShort
            id={id}
            name={name}
            removeRecipe={this.removeRecipe}
            key={id}
            type="favorite"
          />))
        }
      </ul>
    );
  }

  render() {
    return (
      <ManageRecipes>
        <div className="container page-manage__recipes">

          {
            this.renderBody()
          }
          {
            (this.state.last > 1 && this.state.recipes.length !== 0) &&
            <footer>
              <ReactPaginate
                previousLabel={<i className="material-icons">chevron_left</i>}
                nextLabel={<i className="material-icons">chevron_right</i>}
                pageCount={this.state.last}
                marginPagesDisplayed={2}
                pageClassName="waves-effect"
                pageRangeDisplayed={4}
                onPageChange={this.handlePageClick}
                activeClassName="active"
                containerClassName={`pagination ${this.props.isFetching ? 'is-fetching' : ''}`}
              />
            </footer>
          }

        </div>
      </ManageRecipes>
    );
  }
}

FavoriteRecipesPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
};

const mapStateToProps = ({
  favoriteRecipes: { recipes, pagination },
  isFetching,
}) => ({
  recipes,
  pagination,
  isFetching
});

export { FavoriteRecipesPage as PureFavoriteRecipesPage };
export default connect(mapStateToProps)(FavoriteRecipesPage);
