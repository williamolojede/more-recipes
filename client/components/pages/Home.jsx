import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';

import { fetchTopRecipes } from '../../actions/recipe';

import TopRatedRecipeList from '../shared/TopRatedRecipeList';
import Preloader from '../shared/Preloader';
import ConnectedSiteNav from '../shared/SiteNav';
import SiteFooter from '../shared/SiteFooter';
import Search from '../shared/Search';

import { currentUserPropTypes } from '../../config/proptypes';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipes: [],
      currentPage: 1,
      limit: 6,
      last: 1,
      isSearchActive: false
    };
  }

  componentDidMount() {
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  componentWillReceiveProps(newProps) {
    if (newProps.recipes) {
      const {
        recipes,
        pagination
      } = newProps;
      this.setState({
        recipes,
        last: pagination.last
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.currentPage !== this.state.currentPage) {
      this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
    }
  }

  toggleSearch =() => {
    this.setState({
      isSearchActive: !this.state.isSearchActive
    });
  }

  handlePageClick = ({ selected }) => {
    this.setState({
      currentPage: selected + 1
    });
  }

  renderBody = () => {
    const { isFetching } = this.props;

    if (isFetching) {
      return (<Preloader />);
    } else if (!isFetching && this.state.recipes.length === 0) {
      return (
        <p className="no-recipes">
          Sorry no recipes present at the moment
          <span role="img" aria-label="sad">üò¢  </span>, please try again later
          <span role="img" aria-label="please">üôè</span>
        </p>
      );
    }
    return (<TopRatedRecipeList recipes={this.state.recipes} />);
  }

  render() {
    return (
      <div className="page page__home">
        <header className="site-header">
          <ConnectedSiteNav currentUser={this.props.currentUser} />
          <section className="site-header__hero">
            <h2>Delicious recipes, just a search away...</h2>
            <input
              className="site-header__hero--input"
              type="text"
              onFocus={this.toggleSearch}
              placeholder="what would you like to cook?"
            />
          </section>
        </header>
        <main>
          <div className="container">
            <section className="page__home--top-rated">
              <header>
                <h1>Top Rated</h1>
              </header>
              {
                this.renderBody()
              }
              <footer>
                <ReactPaginate
                  previousLabel={<i className="material-icons">chevron_left</i>}
                  nextLabel={<i className="material-icons">chevron_right</i>}
                  pageCount={this.state.last}
                  marginPagesDisplayed={2}
                  pageClassName="waves-effect"
                  pageRangeDisplayed={4}
                  onPageChange={this.handlePageClick}
                  activeClassName="active"
                  containerClassName="pagination"
                />
              </footer>


            </section>
          </div>
        </main>
        {
          this.state.isSearchActive && <Search toggleSearch={this.toggleSearch} />
        }
        <SiteFooter />
      </div>
    );
  }
}

Home.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,

  ...currentUserPropTypes
};

const mapStateToProps = ({
  topRecipes: { recipes, pagination },
  isFetching,
  auth: { currentUser }
}) => ({
  recipes,
  pagination,
  currentUser,
  isFetching
});

export { Home as PureHome };
export default connect(mapStateToProps)(Home);
