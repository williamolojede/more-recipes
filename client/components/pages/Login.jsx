import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';

import { authPropTypes } from '../../config/proptypes';

import Footer from '../shared/Footer.jsx';
import AuthForm from '../shared/AuthForm.jsx';
import Preloader from '../shared/Preloader.jsx';
import ErrorDisplay from '../shared/ErrorDisplay.jsx';

import loginUser from '../../actions/loginUser';
import { removeAuthErrorMessage } from '../../actions/auth';


class Login extends Component {
  componentWillMount() {
    if (this.props.errorMessage) {
      this.props.dispatch(removeAuthErrorMessage());
    }
  }

  /**
   * @param {object} formData -
   * @returns {undefined} - return nothing
   */
  userLogin = (formData) => {
    this.props.dispatch(loginUser(formData));
  }

  render() {
    const {
      isFetching,
      errorMessage,
      isAuthenticated,
      location
    } = this.props;
    const { from } = location.state || { from: { pathname: '/' } };
    if (isAuthenticated) {
      return (
        <Redirect to={from} />
      );
    }

    return (
      <div className="page page__login">
        <header className="header center">
          <a href="/" className="white-text brand-logo">MoreRecipes</a>
        </header>

        <main>
          <div className="container">
            <div className="row">
              <div className="login-card card col s11 m6">
                <div className="card-content">
                  {
                    errorMessage ? <ErrorDisplay message={errorMessage} /> : null
                  }
                  <h1 className="card-title center">Log in to MoreRecipes</h1>
                  <AuthForm type="login" authFormSubmit={this.userLogin} />
                </div>
              </div>
            </div>
            <p className="white-text center">
            Donâ€™t have an account?
              <Link to="/signup" className="white-text"> Signup now</Link>
            </p>
            {
              isFetching ? <Preloader /> : null
            }
          </div>
        </main>

        <Footer />
      </div>
    );
  }
}

Login.defaultProps = {
  // the error message should not be fined by default
  errorMessage: undefined
};

Login.propTypes = authPropTypes;

export const mapStateToProps = ({ auth, isFetching }) => {
  const { isAuthenticated, errorMessage } = auth;
  return {
    isAuthenticated,
    errorMessage,
    isFetching
  };
};

export { Login as PureLogin };
export default connect(mapStateToProps)(Login);
