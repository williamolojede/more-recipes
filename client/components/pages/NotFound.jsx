import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div className="page page__not-found">
    <main>
      <h1>404</h1>
      <p>Sorry, the web page your are looking for is missing
        <span role="img" aria-label="cry">ðŸ˜­</span>
      </p>
      <Link to="/">Go back to the home page</Link>
    </main>
  </div>
);


export default NotFound;
