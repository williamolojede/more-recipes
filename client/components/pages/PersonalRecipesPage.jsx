import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import swal from 'sweetalert';

import decodeToken from '../../utils/decodeToken';

import ManageRecipes from '../shared/ManageRecipes';
import Preloader from '../shared/Preloader';
import RecipeCardShort from '../shared/RecipeCard/RecipeCardShort';

import { fetchPersonalRecipes, deleteRecipe } from '../../actions/recipe';

class PersonalRecipesPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipes: [],
      currentPage: 1,
      limit: 12,
      last: 1,
    };

    this.payload = {
      userId: decodeToken().id,
      page: this.state.currentPage,
      limit: this.state.limit
    };
  }

  componentWillMount() {
    const payload = {
      userId: decodeToken().id,
      page: this.state.currentPage,
      limit: this.state.limit
    };
    // console.log(this.payload);
    this.props.dispatch(fetchPersonalRecipes(payload));
  }

  componentWillReceiveProps(newProps) {
    if (newProps.recipes) {
      const {
        recipes,
        pagination
      } = newProps;
      this.setState({
        recipes,
        last: pagination.last
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.currentPage !== this.state.currentPage) {
      const payload = {
        userId: decodeToken().id,
        page: this.state.currentPage,
        limit: this.state.limit
      };
      this.props.dispatch(fetchPersonalRecipes(payload));
    }
  }

  handlePageClick = ({ selected }) => {
    this.setState({
      currentPage: selected + 1
    });
  }

  removeRecipe = (id) => {
    swal({
      title: 'Are you sure?',
      text: 'Once deleted, you will not be able to recover this recipe!',
      icon: 'warning',
      buttons: true,
      dangerMode: true,
    })
      .then((willDelete) => {
        if (willDelete) {
          return this.props.dispatch(deleteRecipe(id));
        }
        return willDelete;
      })
      .then((willDelete) => {
        if (!willDelete) return;
        const payload = {
          userId: decodeToken().id,
          page: (this.state.recipes.length === 1 && this.state.last !== 1) ?
            this.state.currentPage - 1 : this.state.currentPage,
          limit: this.state.limit
        };
        this.props.dispatch(fetchPersonalRecipes(payload));
      });
  }

  renderBody = () => {
    const { isFetching } = this.props;

    if (isFetching) {
      return (<Preloader />);
    } else if (!isFetching && this.state.recipes.length === 0) {
      return (
        <div className="no-favorites">
          <span>You don&apos;t have any personal recipes yet.</span>
          <span>Share your own culinary creations! Add a recipe</span>
        </div>
      );
    }
    return (
      <ul className="row recipes-list">
        {
          this.state.recipes.map(({ id, name }) => (<RecipeCardShort
            id={id}
            name={name}
            key={id}
            removeRecipe={this.removeRecipe}
            type="personal"
          />))
        }
      </ul>
    );
  }

  render() {
    return (
      <ManageRecipes>
        <div className="container page-manage__recipes">

          {
            this.renderBody()
          }
          {
            (this.state.last > 1 && this.state.recipes.length !== 0) &&
            <footer>
              <ReactPaginate
                previousLabel={<i className="material-icons">chevron_left</i>}
                nextLabel={<i className="material-icons">chevron_right</i>}
                pageCount={this.state.last}
                marginPagesDisplayed={2}
                pageClassName="waves-effect"
                pageRangeDisplayed={4}
                onPageChange={this.handlePageClick}
                activeClassName="active"
                containerClassName={`pagination ${this.props.isFetching ? 'is-fetching' : ''}`}
              />
            </footer>
          }

        </div>
      </ManageRecipes>
    );
  }
}

PersonalRecipesPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
};

const mapStateToProps = ({
  personalRecipes: { recipes, pagination },
  isFetching,
}) => ({
  recipes,
  pagination,
  isFetching
});

export { PersonalRecipesPage as PurePersonalRecipesPage };
export default connect(mapStateToProps)(PersonalRecipesPage);
