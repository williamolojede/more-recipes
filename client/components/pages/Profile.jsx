import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { currentUserPropTypes, notificationPropTypes } from '../../config/proptypes';
import { updateUserProfile } from '../../actions/userProfile';
import defaultConfig from '../../config/default';
import imageUpload from '../../utils/imageUpload';

import ConnectedSiteNav from '../shared/SiteNav.jsx';
import Preloader from '../shared/Preloader.jsx';
import SiteFooter from '../shared/SiteFooter';
import ImageUpload from '../shared/ImageUpload';
import Notification from '../shared/Notification';
import SubmitButton from '../shared/SubmitButton';

class Profile extends Component {
  constructor(props) {
    super(props);

    this.state = {
      user: this.props.currentUser,
      // pending, started, failed, completed
      imageUploadProgress: 'pending',
      submitLoading: false
    };
  }

  componentWillReceiveProps(newProps) {
    if (newProps.currentUser.id !== 0) {
      this.setState({ user: newProps.currentUser });
    }
  }

  imageUploadHandler = (imageInput) => {
    imageUpload.call(
      this,
      imageInput,
      `users/${this.props.currentUser.id}--${Date.now()}`,
      (downloadURL) => {
        const user = Object.assign({}, this.state.user);
        user.imgUrl = downloadURL;
        imageInput.disabled = false;
        this.setState({ user, imageUploadProgress: 'completed' });
      }
    );
  }

  handleUpdateProfile = () => {
    this.setState({ submitLoading: true });
    this.props.dispatch(updateUserProfile(this.state.user))
      .then(() => this.setState({ submitLoading: false }));
  }

  handleOnChange = (event, type) => {
    const user = Object.assign({}, this.state.user);
    user[type] = event.target.value;
    this.setState({ user });
  }

  render() {
    const { currentUser } = this.props;

    const {
      fullname,
      email,
    } = this.state.user;

    return (
      this.props.isFetching ?
        <Preloader />
        :
        <div className="page page__profile">
          <header className="site-header">
            <ConnectedSiteNav currentUser={currentUser} />
          </header>
          <main>
            <div className="container z-depth-1">
              <section className="profile">
                <h3 className="profile__title">Profile</h3>
                <div className="profile__body">
                  <ImageUpload
                    imgUrl={this.state.user.imgUrl}
                    className="profile__image"
                    altText={fullname}
                    imageUploadHandler={this.imageUploadHandler}
                    imageUploadProgress={this.state.imageUploadProgress}
                    defaultImgUrl={defaultConfig.userImgUrl}
                  />
                  <div className="profile__details">
                    <label htmlFor="fullname">
                      Name
                      <input
                        type="text"
                        id="fullname"
                        value={fullname}
                        onChange={e => this.handleOnChange(e, 'fullname')}
                      />
                    </label>
                    <label htmlFor="email">
                      Email
                      <input
                        type="text"
                        id="email"
                        value={email}
                        onChange={e => this.handleOnChange(e, 'email')}
                      />
                    </label>
                    <SubmitButton
                      submitText="Save Changes"
                      handleClick={this.handleUpdateProfile}
                      submitLoading={this.state.submitLoading}
                    />
                  </div>
                </div>

              </section>
            </div>
          </main>
          <Notification
            dispatch={this.props.dispatch}
            notification={this.props.notification}
          />
          <SiteFooter />
        </div>
    );
  }
}

Profile.propTypes = {
  ...currentUserPropTypes,
  isFetching: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  ...notificationPropTypes
};

export const mapStateToProps = ({
  isFetching,
  auth: { currentUser },
  notification
}) => ({
  currentUser,
  isFetching,
  notification
});

export { Profile as PureProfile };
export default connect(mapStateToProps)(Profile);
