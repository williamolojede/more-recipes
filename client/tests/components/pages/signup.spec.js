import React from 'react';
import { PureSignup, mapStateToProps } from '../../../components/pages/Signup';

function setup() {
  const props = {
    dispatch: jest.fn(),
    isFetching: false,
    location: {
      state: {
        from: {}
      }
    },
    isAuthenticated: false
  };
  const wrapper = shallow(<PureSignup {...props} />);

  return {
    wrapper,
    props
  };
}

describe('Signup Page', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch an action to remove error message of initial mounting', () => {
    const { props } = setup();
    const propsWithErrorMessage = { ...props, errorMessage: 'some thing wrong' };
    const wrapper = shallow(<PureSignup {...propsWithErrorMessage} />);
    const dispatchSpy = jest.spyOn(wrapper.instance().props, 'dispatch');
    expect(dispatchSpy).toHaveBeenCalledTimes(1);
  });

  it('should render a preloader', () => {
    const { wrapper } = setup();
    wrapper.setProps({ isFetching: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect to previous page or home if user is already authenticated', () => {
    const { wrapper } = setup();
    wrapper.setProps({ isAuthenticated: true });
    expect(wrapper).toMatchSnapshot();
  });

  describe('userSignup', () => {
    it('should dispatch an action to log user in when form is submitted', () => {
      const { wrapper } = setup();
      const dispatchSpy = jest.spyOn(wrapper.instance().props, 'dispatch');

      wrapper.instance().userSignup({
        email: 'test@test.com',
        password: '123456',
        fullname: 'john bull'
      });
      expect(dispatchSpy).toHaveBeenCalledTimes(1);
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with props required by component from store', () => {
      const store = {
        auth: {
          isAuthenticated: false,
          errorMessage: 'something wrong'
        },
        isFetching: false
      };
      expect(mapStateToProps(store).errorMessage).toBe('something wrong');
      expect(mapStateToProps(store).isFetching).toBe(false);
      expect(mapStateToProps(store).isAuthenticated).toBe(false);
    });
  });
});
