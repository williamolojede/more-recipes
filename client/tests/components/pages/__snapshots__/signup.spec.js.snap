// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page should redirect to previous page or home if user is already authenticated 1`] = `
<Redirect
  push={false}
  to={Object {}}
/>
`;

exports[`Signup Page should render a preloader 1`] = `
<div
  className="page page__register"
>
  <header
    className="header center"
  >
    <a
      className="white-text brand-logo"
      href="/"
    >
      MoreRecipes
    </a>
  </header>
  <main>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="register-card card col s11 m6"
        >
          <div
            className="card-content"
          >
            <h1
              className="card-title center"
            >
              Create a MoreRecipes Account
            </h1>
            <AuthForm
              authFormSubmit={[Function]}
              type="signup"
            />
          </div>
        </div>
      </div>
      <p
        className="white-text center"
      >
        Have an account already?
        <Link
          className="white-text"
          replace={false}
          to="/login"
        >
           Log in now
        </Link>
      </p>
      <Preloader />
    </div>
  </main>
  <Footer />
</div>
`;

exports[`Signup Page should render with right amount of elements 1`] = `
<div
  className="page page__register"
>
  <header
    className="header center"
  >
    <a
      className="white-text brand-logo"
      href="/"
    >
      MoreRecipes
    </a>
  </header>
  <main>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="register-card card col s11 m6"
        >
          <div
            className="card-content"
          >
            <h1
              className="card-title center"
            >
              Create a MoreRecipes Account
            </h1>
            <AuthForm
              authFormSubmit={[Function]}
              type="signup"
            />
          </div>
        </div>
      </div>
      <p
        className="white-text center"
      >
        Have an account already?
        <Link
          className="white-text"
          replace={false}
          to="/login"
        >
           Log in now
        </Link>
      </p>
    </div>
  </main>
  <Footer />
</div>
`;
